ibm/service/kubernetes/resource_ibm_container_dedicated_host.go

```diff
@@ -0,0 +1,482 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	dedicatedHostAPI := client.DedicatedHost()
+	targetEnv := v2.ClusterTargetHeader{}
+
+	hostPoolID := d.Get("host_pool_id").(string)
+
+	params := v2.CreateDedicatedHostRequest{
+		Flavor:     d.Get("flavor").(string),
+		HostPoolID: hostPoolID,
+		Zone:       d.Get("zone").(string),
+	}
+
+	res, err := dedicatedHostAPI.CreateDedicatedHost(params, targetEnv)
+	if err != nil {
+		return diag.Errorf("[ERROR] CreateDedicatedHost failed: %v", err)
+	}
+	hostID := res.ID
+	d.SetId(fmt.Sprintf("%s:%s", hostPoolID, hostID))
+
+	dh, err := waitForDedicatedHostAvailable(ctx, dedicatedHostAPI, hostID, hostPoolID, DedicatedHostCreateTimeout, targetEnv)
+	if err != nil {
+		return diag.Errorf("[ERROR] waitForDedicatedHostAvailable failed: %v", err)
+	}
+
+	dedicatedHost, ok := dh.(v2.GetDedicatedHostResponse)
+	if !ok {
+		return diag.Errorf("[ERROR] waitForDedicatedHostAvailable response is faulty: %v", dh)
+	}
+
+	setDedicatedHostFields(d, dedicatedHost)
+
+	placement, ok := d.GetOk("placement_enabled")
+	if ok && dedicatedHost.PlacementEnabled != placement.(bool) {
+		req := v2.UpdateDedicatedHostPlacementRequest{
+			HostPoolID: hostPoolID,
+			HostID:     hostID,
+		}
+		if placement.(bool) {
+			if err = dedicatedHostAPI.EnableDedicatedHostPlacement(req, targetEnv); err != nil {
+				return diag.Errorf("[ERROR] EnableDedicatedHostPlacement failed: %v", err)
+			}
+		} else {
+			if err = dedicatedHostAPI.DisableDedicatedHostPlacement(req, targetEnv); err != nil {
+				return diag.Errorf("[ERROR] DisableDedicatedHostPlacement failed: %v", err)
+			}
+		}
+		_, err = waitForDedicatedHostPlacement(ctx, dedicatedHostAPI, hostID, hostPoolID, placement.(bool), DedicatedHostCreateTimeout, targetEnv)
+		if err != nil {
+			return diag.Errorf("[ERROR] waitForDedicatedHostPlacement failed: %v", err)
+		}
+		d.Set("placement_enabled", placement)
+	}
+
+	return nil
+}
+
+func resourceIBMContainerDedicatedHostRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	id := d.Id()
+	if err := getIBMContainerDedicatedHost(id, d, meta); err != nil {
+		return diag.Errorf("[ERROR] getIBMContainerDedicatedHost failed: %v", err)
+	}
+	return nil
+}
+
+func getIBMContainerDedicatedHost(id string, d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(conns.ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+	dedicatedHostAPI := client.DedicatedHost()
+	targetEnv := v2.ClusterTargetHeader{}
+
+	// <hostpoolid>:<hostid>
+	m := strings.Split(id, ":")
+	if len(m) < 2 || m[0] == "" || m[1] == "" {
+		return fmt.Errorf("[ERROR] unexpected format of ID (%s), the expected format is <hostpoolid>:<hostid>", id)
+	}
+	hostPoolID := m[0]
+	hostID := m[1]
+
+	dedicatedHost, err := dedicatedHostAPI.GetDedicatedHost(hostID, hostPoolID, targetEnv)
+
+	if err != nil {
+		if apiErr, ok := err.(bmxerror.RequestFailure); ok {
+			if apiErr.StatusCode() == 404 {
+				return nil
```

Comment:
If its 404 then d.setID("") and then return

Reply:
done